src/components/dropdownSearch/MultiSelectDropdownSearch/MultiSelectDropdownSearchReducer.tsx

```diff
@@ -0,0 +1,134 @@
+import { DropdownSearchActions, IOptionsDropdownSearchPayload } from '../DropdownSearchActions';
+import {
+  addUniqueSelectedOption,
+  dropdownSearchInitialState,
+  dropdownSearchReducer,
+  getDisplayedOptions,
+  getNextIndexPosition,
+  getOptionsFiltered,
+  IDropdownSearchState,
+  removeSelectedOption,
+} from '../DropdownSearchReducers';
+import { IReduxAction } from '../../../utils/ReduxUtils';
+import {FixedQueue} from '../../../utils/FixedQueue';
+import {IDropdownOption} from '../DropdownSearch';
+import {keyCode} from '../../../utils/InputUtils';
+
+export const multiSelectDropdownSearchReducer = (state: IDropdownSearchState = dropdownSearchInitialState,
+  action: IReduxAction<IOptionsDropdownSearchPayload>) => {
+
+  switch (action.type) {
+    case DropdownSearchActions.addMultiSelect:
+      return {
+        ...state,
+        id: action.payload.id,
+        options: action.payload.optionsDropdown,
+        selectedOptions: new FixedQueue<IDropdownOption>(),
+        filterText: action.payload.filterText,
+        isOpened: false,
+      };
+    case DropdownSearchActions.removeSelectedOption:
+      state.selectedOptions = removeSelectedOption(state, action.payload.selectedOptionValue);
+      state.displayedOptions = getDisplayedOptions(state);
+      return {
+        ...state,
+        id: action.payload.id,
+        selectedOptions: state.selectedOptions,
```

Comment:
For any other property that is not an object or an array, you could have simply removed this line since it is included in the `...state`, but since `selectedOptions` is an array, you will need to copy its content.
```
selectedOptions: [
  ...state.selectedOptions,
],
```
If the elements inside `selectedOptions` are objects, you will need to copy them too. If that is the case, I would recommend you only save their ids.

Same thing applies for `displayedOptions`